src/link/interface.ts

```diff
@@ -0,0 +1,57 @@
+/* eslint-disable @typescript-eslint/no-unnecessary-type-constraint */
+/* eslint-disable no-use-before-define */
+import type { MultihashDigest } from '../hashes/interface'
+import type { MultibaseEncoder, MultibaseDecoder, Multibase } from '../bases/interface'
+import type { Phantom, ByteView } from '../block/interface'
+
+
+export type { MultihashDigest, MultibaseEncoder, MultibaseDecoder }
+export type Version = 0 | 1
+
+export type DAG_PB = 0x70
+export type SHA_256 = 0x12
+
+/**
+ * Represents an IPLD link to a specific data of type `T`.
+ *
+ * @template T - Logical type of the data being linked to.
+ * @template C - multicodec code corresponding to a codec linked data is encoded with
+ * @template A - multicodec code corresponding to the hashing algorithm of the CID
+ * @template V - CID version
+ */
+export interface Link<
+  Data extends unknown = unknown,
+  Format extends number = number,
+  Alg extends number = number,
+  V extends Version = 1
+  > extends Phantom<Data> {
+  readonly version: V
+  readonly code: Format
+  readonly multihash: MultihashDigest<Alg>
+
+  readonly byteOffset: number
+  readonly byteLength: number
+  readonly bytes: ByteView<Link<Data, Format, Alg, V>>
+
+
+  equals(other: unknown): other is Link<Data, Format, Alg, Version>
+
+  toString<Prefix extends string>(base?: MultibaseEncoder<Prefix>): ToString<Link<Data, Format, Alg, Version>, Prefix>
+  toJSON(): { version: V, code: Format, hash: Uint8Array }
+  link(): Link<Data, Format, Alg, V>
+
+  toV1(): Link<Data, Format, Alg, 1>
+}
+
+
+
+export interface LegacyLink<T extends unknown = unknown> extends Link<T, DAG_PB, SHA_256, 0> {
+}
+
+export type UnknownLink =
+  | LegacyLink<unknown>
+  | Link<unknown, number, number, Version>
+
+export type ToString<T, Prefix extends string = string> = Multibase<Prefix> & Phantom<T>
+
+export type { ByteView }
```

Comment:
Is this a necessary export? I think this interface is mostly used through the `API` forms of it in ../interface which already pulls in ../block/interface that has this.

Reply:
It’s just nice to have all the link related types in one place so you don’t have to import two modules later. I have bunch of places where I’d import just this module if I could access this type and Link